### PR TITLE
Plots refactor to use Vue and es6 instead of angular

### PR DESCRIPTION
**Pull request by shefalijoshi**
Single, Overlay and Stacked plots as Vue components
Known issues: #3626, #3631, #3655, #3687

Resolves: #3691

### Author Checklist

1. Changes address original issue? Y
2. Unit tests included and/or updated with changes? Y
3. Command line build passes? Y
4. Changes have been smoke-tested? Y
5. Testing instructions included? Y

### CODE REVIEW

File: index.html
Comment (shefalijoshi) [inline index.html:90]: This is temporary - eventually we will only use this in the Timeline.

File: src/plugins/plot/plugin.js
Comment (shefalijoshi) [inline src/plugins/plot/plugin.js:88]: This is temporary - we will put it back when WIP is done.

**Comment by deeptailor** [comment]
This looks great! Just want to add a couple of improvements (from our old plots):

1. For the plot legend, if we can re-use telemetry tables, that would be great. It will allow users to set column widths and expand columns etc, which has been a much requested feature.
2. For plot zoom via mousewheel, we should use pinch zoom in / out as well, user's have complained that zooming via a trackpad does not work as expected. For some reason, mousewheel zoom in equals pinch zoom out (browser inconsistency).

I will add more as I think of them. But this is really great so far!

File: src/plugins/plot/vue/single/MctPlot.vue
Comment (nikhilmandlik) [inline src/plugins/plot/vue/single/MctPlot.vue:165]: remove extra space 😃

File: src/plugins/plot/vue/single/MctPlot.vue
Comment (nikhilmandlik) [inline src/plugins/plot/vue/single/MctPlot.vue:168]: same

File: src/plugins/plot/vue/single/MctPlot.vue
Comment (nikhilmandlik) [inline src/plugins/plot/vue/single/MctPlot.vue:294]: wondering if we can call 'initCanvas' here as it is doing same things.

File: src/plugins/plot/vue/single/configuration/Model.js
Comment (nikhilmandlik) [inline src/plugins/plot/vue/single/configuration/Model.js:56]: empty?

**Review by nikhilmandlik — COMMENTED** [review]
looking great 👍, Small changes and needs some cleanup of usual suspects of spacing and ordering methods 😃

**Comment by jvigliotta** [comment]
Something to look out for during refactor: #3599 

Example Imagery is able to be dropped onto a plot

**Comment by shefalijoshi** [comment]
> This looks great! Just want to add a couple of improvements (from our old plots):
> 
>     1. For the plot legend, if we can re-use telemetry tables, that would be great. It will allow users to set column widths and expand columns etc, which has been a much requested feature.
> 
>     2. For plot zoom via mousewheel, we should use pinch zoom in / out as well, user's have complained that zooming via a trackpad does not work as expected. For some reason, mousewheel zoom in equals pinch zoom out (browser inconsistency).
> 
> 
> I will add more as I think of them. But this is really great so far!

@deeptailor I'll add an issue to use tables for the legend. So that this PR doesn't get even bigger.

**Comment by shefalijoshi** [comment]
@charlesh88 I need some help with css to tweak the gridlines in plots with this refactor.

File: src/plugins/plot/vue/single/MctTicks.vue
Comment (deeptailor) [inline src/plugins/plot/vue/single/MctTicks.vue:239]: Angular had some weird irregularities. We wont need any forced digests with vue

File: src/plugins/plot/vue/single/Plot.vue
Comment (deeptailor) [inline src/plugins/plot/vue/single/Plot.vue:105]: No need for this comment, the method is self explanatory

File: src/plugins/plot/vue/single/Plot.vue
Comment (deeptailor) [inline src/plugins/plot/vue/single/Plot.vue:114]: same as above

**Review by deeptailor — CHANGES_REQUESTED** [review]
This looks great, just did a quick preliminary review with a few comments

File: src/plugins/plot/vue/single/PlotViewProvider.js
Comment (shefalijoshi) [reply src/plugins/plot/vue/single/PlotViewProvider.js:50]: ok done!

File: src/plugins/plot/vue/single/MctPlot.vue
Comment (shefalijoshi) [reply src/plugins/plot/vue/single/MctPlot.vue:25]: Done

File: src/plugins/plot/vue/single/MctPlot.vue
Comment (shefalijoshi) [reply src/plugins/plot/vue/single/MctPlot.vue:226]: Good question! I've removed that line and we're just using this.domainObject now.

File: src/plugins/plot/vue/single/MctTicks.vue
Comment (shefalijoshi) [reply src/plugins/plot/vue/single/MctTicks.vue:239]: Yeah, I'll remove those comments.

File: src/plugins/plot/vue/single/MctPlot.vue
Comment (shefalijoshi) [reply src/plugins/plot/vue/single/MctPlot.vue:876]: Hmm. Adding these to the eventHelper listenTo doesn't quite fit since the helper mostly manages listeners on specific objects but the listening to domain object changes via objects.observe api is a bit different.
I'll think about it again, but right now it doesn't feel right.

File: src/plugins/plot/vue/single/MctTicks.vue
Comment (shefalijoshi) [reply src/plugins/plot/vue/single/MctTicks.vue]: Done.

**Review by akhenry — COMMENTED** [review]
Have just started looking at this. Looks great so far. An initial general comment, and it's a small thing, we don't need the `Mct` prefix on Vue components. This was a convention we used in the Angular days to avoid namespace collisions with our directives, but we don't need to do it any more.

Will provide some more detailed feedback as I get further into it.

**Review by akhenry — CHANGES_REQUESTED** [review]

File: src/plugins/plot/vue/single/draw/DrawWebGL.js
Comment (akhenry) [inline src/plugins/plot/vue/single/draw/DrawWebGL.js:85]: Comment seems to be duped.

File: src/plugins/plot/vue/single/MctPlot.vue
Comment (shefalijoshi) [reply src/plugins/plot/vue/single/MctPlot.vue:424]: Created a new issue for this #3655

File: src/plugins/plot/vue/single/MctPlot.vue
Comment (shefalijoshi) [reply src/plugins/plot/vue/single/MctPlot.vue:228]: The userViewportChangeEnd event no longer needs to be emitted. I'm going to get rid of those.

Comment (shefalijoshi) [reply src/plugins/plot/vue/single/MctPlot.vue:230]: I have added a note as to why $nextTick is needed.

File: src/plugins/plot/vue/single/LinearScale.js
Comment (shefalijoshi) [reply src/plugins/plot/vue/single/LinearScale.js:28]: Done.

**Comment by akhenry** [comment]
Some minor things I discovered while testing this:
* plot grids do not show in the new Vue plots
* The cursor should change from a cross to a hand when the pan modifier key is held down (same as conductor axis)

Neither of these need to be addressed before this is merged though, we don't need them for timeline plots. We should file a followup issue and collect anything that needs to be addressed to reach parity with Angular plots.

**Review by akhenry — APPROVED** [review]

**Comment by akhenry** [comment]
@shefalijoshi  Looks like we just need some tests to get us back up to 66% coverage....Adding a plugin spec that just installs the plugin will probably get us most of the way there tbh.

**Comment by akhenry** [comment]
followup issues tracked here - https://github.com/nasa/openmct/issues/3687

**Comment by akhenry** [comment]
@shefalijoshi Could you please link to an issue, and provide an author checklist.

@deeptailor Could you take a look at your review comments and approve if they've been addressed?

**Comment by akhenry** [comment]
@shefalijoshi  Also, don't worry about writing test instructions specifically for this PR, just include a step on the timeline issue to add a plot to a timeline.

**Review by deeptailor — APPROVED** [review]
LGTM!

**Comment by deeptailor** [comment]
## Reviewer Checklist
Changes appear to address issue? Yes
Appropriate unit tests included? Yes
Code style and in-line documentation are appropriate? Yes
Commit messages meet standards? Yes
Has associated issue been labelled unverified? (only applicable if this PR closes the issue) Yes

**Pull request merged**
